### DISCUSSION TITLE
Wiresharking to discover "Lead" track for StateMap?

### DISCUSSION THREAD

**Discussion by chrisle** [icedream/go-stagelinq#15] [Q&A] [answered]
There is state for "Lead" track (ie: which one is currently playing / similar to ProDJ-Link "on-air" state).

I was wondering, is there any guidance (even if it's basic) on sniffing / wiresharking stagelinq to see if I can discover new state values? I'm hoping to uncover a the "lead" track state and contribute to this and other projects.

**Comment by icedream** [comment, marked as answer]
**EDIT: Since it's answered in a comment under my reply, I'll refer to that and mark this as answer. GitHub does not allow marking the reply as an answer unfortunately.**

Personally, I discovered state values by using preexisting software that speaks the protocol to see which states it subscribes to. However that's definitely not the way to find *all* state values, there's always a possibility that there's more of them.

**Reply by chrisle** [reply]
Well I ended up buying an X1850, two SC6000s, and a Hak5 Packet Squirrel. Grabbed some PCAP files and found the answer to this one. The answer appears to be /Client/Deck1/DeckIsMaster. Here are a few more I found in the meantime:

```
ClientPreferencesLayerA           = "/Client/Preferences/LayerA"
ClientPreferencesPlayerJogColorA  = "/Client/Preferences/PlayerJogColorA"
ClientPreferencesPlayerJogColorB  = "/Client/Preferences/PlayerJogColorB"
EngineDeck1DeckIsMaster           = "/Client/Deck1/DeckIsMaster"
EngineDeck2DeckIsMaster           = "/Client/Deck2/DeckIsMaster"
EngineMasterMasterTempo           = "/Engine/Master/MasterTempo"
EngineSyncNetworkMasterStatus     = "/Engine/Sync/Network/MasterStatus"
MixerChannelAssignment1           = "/Mixer/ChannelAssignment1"
MixerChannelAssignment2           = "/Mixer/ChannelAssignment2"
MixerChannelAssignment3           = "/Mixer/ChannelAssignment3"
MixerChannelAssignment4           = "/Mixer/ChannelAssignment4"
MixerNumberOfChannels             = "/Mixer/NumberOfChannels"
```
